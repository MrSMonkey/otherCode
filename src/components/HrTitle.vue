/*
 * @Description: 两侧带横线的标题组件
 * @Author: chenmo
 * @Date: 2019-02-15 10:27:12
 * @Last Modified by: chenmo
 * @Last Modified time: 2019-02-15 11:15:42
 */

<template>
  <section class="hr-title" ref="header">
    <span class="line"></span>
    <h2 class="content">
      {{title}}
      <span>{{dec}}</span>
    </h2>
    <span class="line"></span>
  </section>
</template>

<script lang="ts">
import { Component, Vue, Prop } from 'vue-property-decorator';

@Component({
  name: 'hrTitle',
})
export default class Header extends Vue {
  @Prop({ type: String, default: 'Title' })
  private title: any;
  @Prop({ type: String, default: '' })
  private dec: any;
}
</script>

<style lang="stylus" rel="stylesheet/stylus">
@import '../assets/stylus/index.styl'

.hr-title
  width 100%
  height 36px
  text-align center
  display flex
  justify-content: space-between
  align-items center
  padding 0 vw(20)
  .content
    font-size 16px
    color $text-color
    span {
      display block
      font-size 12px
      font-weight lighter
      color $next-text-color
    }
  .line 
    display  inline-block
    width 80px
    border-top 1px solid $disabled-color
</style>
