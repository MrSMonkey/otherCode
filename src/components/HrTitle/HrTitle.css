@import '../../styles/variables.global.css';

.hr-title {
  padding: 0 30px;
  width: 100%;
  text-align: center;
  display: flex;
  justify-content: space-between;
  align-items: center;
  font-size: 18px; /* no */
  h2 {
    font-size: 18px; /* no */
     span {
      display: block;
      font-size: 12px; /* no */
      color: $font-color-light;
      font-weight: lighter;
    }
  }
}

.hr-line {
  width: 170px;
  height: 0;
  border-top: 1px solid $border-color-default; /* no */
}