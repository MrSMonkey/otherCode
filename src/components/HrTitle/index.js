import React from 'react'
import CSSModules from 'react-css-modules'
import styles from './HrTitle.css'

const HrTitle = (props) => (
  <div styleName="hr-title">
    <div styleName="hr-line"></div>
    <h2>
      {props.title}
      <span>{props.dec}</span>
    </h2>
    <div styleName="hr-line"></div>
  </div>
)

export default CSSModules(HrTitle, styles);