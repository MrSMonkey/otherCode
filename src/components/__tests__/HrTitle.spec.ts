/*
 * @Description: 底部按钮单元组件测试
 * @Author: chenmo
 * @Date: 2019-03-26 15:28:22
 * @Last Modified by: chenmo
 * @Last Modified time: 2019-03-27 11:51:02
 */

import Vue from 'vue';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router';
import HrTitle from '@/components/HrTitle.vue';

// 独立构建Vue构造函数
const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(VueRouter);

describe('对HrTitle组件进行的单元测试', () => {
  let store: any;
  let wrapper: any;
  let propsData: any;

  beforeEach(() => {
    propsData = {
      title: '标题',
      dec: '描述'
    };

    store = new Vuex.Store({
      state: {
        global: {}
      }
    });

    // 挂载组件
    wrapper = shallowMount(HrTitle, {
      store,
      propsData,
      localVue
    });
  });

  afterEach(() => {
    // 销毁实例
    wrapper.destroy();
  });

  test('BottomBtn', () => {
    // 测试实例是Vue组件
    expect(wrapper.isVueInstance()).toBeTruthy();
    // 存在 BottomBtn 组件
    expect(wrapper.is(HrTitle)).toBeTruthy();
    // 验证组件是否渲染
    expect(wrapper.contains('.hr-title')).toBeTruthy();
  });

  // 生成测试快照
  test('create expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
