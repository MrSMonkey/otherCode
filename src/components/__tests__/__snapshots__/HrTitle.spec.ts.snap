// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`对HrTitle组件进行的单元测试 create expected html structure 1`] = `
<section
  class="hr-title"
>
  <span
    class="line"
  />
   
  <h2
    class="content"
  >
    
    标题
    
    <span>
      描述
    </span>
  </h2>
   
  <span
    class="line"
  />
</section>
`;
